app/models/media_object.rb

```diff
@@ -569,7 +563,48 @@ def move_bulk documents, params
 
   end
 
+  def _update_permalink_and_dependents
+    ensure_permalink!
+    self.parts.each do |master_file| 
+      begin
+        updated = master_file.ensure_permalink!
+        master_file.save( validate: false ) if updated
+      rescue
+        # no-op
+        # Save is called (uncharacteristically) during a destroy.
+      end
+    end
+
+    unless self.descMetadata.permalink.include? self.permalink 
+      self.descMetadata.permalink = self.permalink
+    end
+  end
+  handle_asynchronously :_update_permalink_and_dependents
+  
+  def update_permalink_and_dependents
+    self._update_permalink_and_dependents
+  end
+
+  def _remove_bookmarks
```

Comment:
Right. I had it running asynchronously, because if there are a lot of bookmarks it could take some time, but it was causing other tests to fail. I left it broken out in case we wanted to revisit it later, but I'm happy to remove the indirection if you think it's confusing.